sklearn/ensemble/tests/test_bagging.py

```diff
@@ -663,3 +663,64 @@ def test_oob_score_removed_on_warm_start():
     clf.fit(X, y)
 
     assert_raises(AttributeError, getattr, clf, "oob_score_")
+
+
+def test_consistent_index_sampling():
+    # Make sure randomly drawn indices generated at fit time are identically 
+    # reproduced at a later time (for use in, e.g., OOB scoring)
+    X, y = make_hastie_10_2(n_samples=20, random_state=1)
+    
+    bagging = BaggingClassifier(KNeighborsClassifier(), max_samples=0.5,
+                                max_features=0.5)
+    bagging.fit(X, y)
+
+    feat_inds_list = bagging.estimators_features_
+    feat_samp_inds_gen = bagging._get_estimators_indices()
+
+    for feat_inds1, (feat_inds2, _) in zip(feat_inds_list, feat_samp_inds_gen):
+        assert_array_equal(feat_inds1, feat_inds2) 
+
+
+def test_oob_score_consistency():
+    # Make sure OOB scores are identical when random_state, estimator, and 
+    # training data are fixed and fitting is done twice
+    X, y = make_hastie_10_2(n_samples=200, random_state=1)
+    bagging = BaggingClassifier(KNeighborsClassifier(), max_samples=0.5,
+                                max_features=0.5, oob_score=True,
+                                random_state=1)
+    assert_equal(bagging.fit(X, y).oob_score_, bagging.fit(X, y).oob_score_)
+
+
+def test_estimators_samples():
+    # Check that format of estimators_samples_ is correct and that results
+    # generated at fit time can be identically reproduced at a later time
+    # using data saved in object attributes.
+    X, y = make_hastie_10_2(n_samples=200, random_state=1)
+    bagging = BaggingClassifier(LogisticRegression(), max_samples=0.5,
+                                max_features=0.5, random_state=1,
+                                bootstrap=False)
+    bagging.fit(X, y)
+
+    # Get attributes, convert possible generators to lists
+    estimators_samples = list(bagging.estimators_samples_)
+    estimators_features = bagging.estimators_features_
+    estimators = bagging.estimators_
+
+    # Test for correct formatting
+    assert_equal(len(estimators_samples), len(estimators))
+    assert_equal(len(estimators_samples[0]), len(X))
+    assert_true(isinstance(estimators_samples[0][0], (np.bool_, bool)))
```

Comment:
the length thing is fair enough. otherwise, `assert_equal(arr.dtype.kind, 'b')` might be easiest